resources/sts/4.14/hypershift/sts_extended_hcp_support_permission_policy.json

```diff
@@ -62,49 +137,300 @@
+        },
+        {
+            "Sid": "MulticastDomain",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:GetTransitGatewayMulticastDomainAssociations",
+                "ec2:SearchTransitGatewayMulticastGroups"
+            ],
+            "Resource": "arn:aws:ec2:*:*:transit-gateway-multicast-domain/*"
+        },
+        {
+            "Sid": "TransitGatewayRoutes",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:SearchTransitGatewayRoutes"
+            ],
+            "Resource": "arn:aws:ec2:*:*:transit-gateway-route-table/*"
+        },
+        {
+            "Sid": "Vpc",
+            "Effect": "Allow",
+            "Action": [
                 "ec2:DescribeVpcClassicLink",
                 "ec2:DescribeVpcClassicLinkDnsSupport",
                 "ec2:DescribeVpcEndpointConnectionNotifications",
                 "ec2:DescribeVpcEndpointServiceConfigurations",
-                "ec2:DescribeVpcEndpointServicePermissions",
                 "ec2:DescribeVpcEndpointServices",
-                "ec2:DescribeVpcPeeringConnections",
-                "ec2:DescribeVpnConnections",
-                "ec2:DescribeVpnGateways",
-                "ec2:GetAssociatedIpv6PoolCidrs",
-                "ec2:GetManagedPrefixListEntries",
-                "ec2:GetSerialConsoleAccessStatus",
-                "ec2:GetTransitGatewayAttachmentPropagations",
-                "ec2:GetTransitGatewayMulticastDomainAssociations",
-                "ec2:GetTransitGatewayPrefixListReferences",
-                "ec2:GetTransitGatewayRouteTableAssociations",
-                "ec2:GetTransitGatewayRouteTablePropagations",
-                "ec2:ModifyInstanceAttribute",
-                "ec2:SearchLocalGatewayRoutes",
-                "ec2:SearchTransitGatewayMulticastGroups",
-                "ec2:SearchTransitGatewayRoutes",
-                "ec2:StartNetworkInsightsAnalysis",
-                "elasticloadbalancing:ConfigureHealthCheck",
-                "kms:CreateGrant",
-                "s3:GetBucketTagging",
+                "ec2:DescribeVpcPeeringConnections"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "VpcAttribute",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribeVpcAttribute"
+            ],
+            "Resource": "arn:aws:ec2:*:*:vpc/*"
+        },
+        {
+            "Sid": "VpcEndpointService",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribeVpcEndpointServicePermissions"
+            ],
+            "Resource": "arn:aws:ec2:*:*:vpc-endpoint-service/*"
+        },
+        {
+            "Sid": "S3object",
+            "Effect": "Allow",
+            "Action": [
                 "s3:GetObjectAcl",
-                "s3:GetObjectTagging",
-                "s3:ListAllMyBuckets",
+                "s3:GetObjectTagging"
+            ],
+            "Resource": "arn:aws:s3:::*"
```

Comment:
What S3 buckets are these actions used for? Is there any way to scope this down to only those S3 buckets?

Reply:
Looks like Image-registry create a bucket upon cluster creation '29hbfpnu5een3jvr60o7sj0uu8knd475-image-registry-us-east-1-guyi'

Will add condition red-hat-managed:true to this